### ISSUE TITLE
App crashed

### ISSUE THREAD

**Issue by elysdir** [gwillen/solstice-audio-test#65]
Hi--I pressed the start button to start the audio test, and I got the following message:

This app has crashed. We're really sorry :-(
Please file a bug with the following information; it will help us fix it.

TypeError: Failed to construct 'MediaStreamAudioSourceNode': required member mediaStream is undefined.
TypeError: Failed to construct 'MediaStreamAudioSourceNode': required member mediaStream is undefined.
    at EventTarget.start_bucket (https://echo.jefftk.com/app.js:562:20)
    at start (https://echo.jefftk.com/demo.js:940:20)
    at async HTMLButtonElement.start_stop (https://echo.jefftk.com/demo.js:483:5)

I'll try again.

(I'm on a Mac (macOS 10.15.7), using Chrome (Version 87.0.4280.67 (Official Build) (x86_64)).

**Comment by gwillen** [comment]
You're the second person to report this error (unless that was also you the other day.) The first one didn't give any details, but said something about permissions. Is it possible you pressed 'deny' to the microphone permissions dialog by accident (or closed it without pressing either?)

Obviously we should handle that better, if that's what happened.

**Comment by jeffkaufman** [comment]
I wonder if this might also happen if someone ignores the permissions pop up?

**Comment by elysdir** [comment]
I don't know what happened, but yeah, my mic was disallowed somehow. I tried various things; eventually reset the Site Settings and reloaded the page, and it worked. Thank you!

**Comment by elysdir** [comment]
Followup clarification, ’cause I was in a rush when I wrote that last response:

Usually I’m good about allowing mic permissions, but I think you’re right that I must have accidentally clicked Deny.

After I saw your responses, I tried going in and changing the permissions to Allow, but it still didn’t seem to work. (But I may not have tested it thoroughly at that point.) But then I clicked the reset-permissions button in Chrome’s “Site Settings” screen, and then reloaded the page and everything worked.

Maybe one way for the system to detect this state would be to check whether the dropdown menu for “Input device” only has “Unknown Input”. I didn’t know what that dropdown was supposed to say, so I didn’t take “Unknown Input” as a warning sign; but maybe a short-term quasi-fix would be to add a note to tell the user that if the dropdown says “Unknown Input,” then they need to reset their mic permissions in Site Settings. ...Anyway, just a thought; feel free to ignore that.

Thanks for figuring out the problem! And thank you for creating the system and making it available!
